fulp_modules/minebot_improvements/minebot_additions.dm

```diff
@@ -0,0 +1,95 @@
+	if ("lava" in M.weather_immunities)
+		to_chat(user, "<span class='warning'>[M] already has lava-proof plating installed!</span>")
+		return
+	M.weather_immunities += list("lava")
+	to_chat(user, "<span class='notice'>You apply the lava-proof plating on [M].</span>")
+	qdel(src)
+
+
+// Speed Upgrade
+/obj/item/mine_bot_upgrade/speed
+	name = "minebot speed upgrade"
+
+/obj/item/mine_bot_upgrade/speed/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	if(M.cached_multiplicative_slowdown != 2.5)  //Checks for the current slowdown of the Minebot, to see if it's not different from the default value, which is 2.5
+		to_chat(user, "<span class='warning'>[M] already has a speed upgrade installed!</span>")
+		return
+	M.add_movespeed_modifier(/datum/movespeed_modifier/minebot_speedupgrade)  //This makes it so a normal miner would still go twice as fast, but this would still be a significant speed upgrade for the Minebots, going from a slowdown of 3 to a slowdown of 2.
+	to_chat(user, "<span class='notice'>You apply the speed upgrade on [M].</span>")
+	qdel(src)
+
+
+// Name Change
+/obj/item/mine_bot_upgrade/renaming
+	name = "minebot reclassification board"
+	desc = "A Minebot upgrade that allows you to rename your Minebot!"
+	icon_state = "door_electronics"
+	icon = 'icons/obj/module.dmi'
+
+	var/being_used = FALSE
+
+/obj/item/mine_bot_upgrade/renaming/attack(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	. = ..()
+	if(being_used || !istype(M, /mob/living/simple_animal/hostile/mining_drone))
+		return
+	being_used = TRUE
+
+	to_chat(user, "<span class='notice'>You start changing your Minebot's name...</span>")
+
+	var/new_name = stripped_input(user, "What would you like your Minebot's new name to be?", "Input a name", M.real_name, MAX_NAME_LEN)
+
+	if(!new_name || QDELETED(src) || QDELETED(M) || new_name == M.real_name || !M.Adjacent(user))
+		being_used = FALSE
+		return
+
+	M.visible_message("<span class='notice'><span class='name'>[M]</span> has a new name, <span class='name'>[new_name]</span>.</span>", "<span class='notice'>Your old name of <span class='name'>[M.real_name]</span> fades away, and your new name <span class='name'>[new_name]</span> anchors itself in your mind.</span>")
+	message_admins("[ADMIN_LOOKUPFLW(user)] used [src] on [ADMIN_LOOKUPFLW(M)], renaming them into [new_name].")
+	var/new_tag = "[new_name] - Minebot"
+	SEND_SIGNAL(M, COMSIG_SIGNAL_SOURCE_RENAMED, new_tag) // Using a variable here because it didn't work as intended when I put the string directly in it
+
+	// pass null as first arg to not update records or ID/PDA
+	M.fully_replace_character_name(null, new_name)
+
+	qdel(src)
+
+
+// Here's the movespeed_modifier that goes for the minebot's speed upgrade
+/datum/movespeed_modifier/minebot_speedupgrade
+	variable = TRUE
+	multiplicative_slowdown = -0.5
+
+
+// Finally, here's how you add to the mining vendor for the new upgrade boards.
```

Comment:
This will conflict with digi shoes PR because you're adding it in some random hidden file rather than a vending_innit.dm file where people can easily find it.

Reply:
Yes, please make this its own file, as I imagine it will be used frequently.

Reply:
I mean, at this point, if that's what you want me to do, just get that thing merged first, so it's not as much of a pain for them to fix it if I do it as well.

Reply:
Like, I did it there because it didn't feel overly necessary at the time, and I couldn't exxactly predict it would take almost two months to get it merged. I'll just wait for the digi shoes to get merged and I'll update this, it's going to be less of a headache.

Reply:
it's merged now, you can use the new file.

Reply:
> it's merged now, you can use the new file.

I'll do it tomorrow then, god damn GitHub sending me notifications three days late.